app/features/idl/interactive-idl/ui/ArgumentInput.tsx

```diff
@@ -2,17 +2,251 @@ import type { ArgField } from '@entities/idl';
 import { Badge } from '@shared/ui/badge';
 import { Input } from '@shared/ui/input';
 import { Label } from '@shared/ui/label';
-import { forwardRef, useId } from 'react';
+import { forwardRef, useEffect, useId, useRef } from 'react';
+import { X } from 'react-feather';
 
-import { isRequiredArg } from '../lib/instruction-args';
+import { getArrayMaxLength, isArrayArg, isRequiredArg, isVectorArg } from '../lib/instruction-args';
+
+const MAX_ARRAY_INPUTS = 100;
 
 export interface ArgumentInputProps extends React.ComponentProps<'input'> {
     arg: ArgField;
-    error: { message?: string | undefined } | undefined;
+    error?: { message?: string | undefined } | undefined;
 }
-
 export const ArgumentInput = forwardRef<HTMLInputElement, ArgumentInputProps>(({ arg, error, ...props }, ref) => {
     const inputId = useId();
+    const isArrayOrVec = isArrayArg(arg) || isVectorArg(arg);
+
+    if (isArrayOrVec) {
+        return <ArrayArgumentInput ref={ref} arg={arg} error={error} inputId={inputId} {...props} />;
+    }
+
+    return <SingleArgumentInput ref={ref} arg={arg} error={error} inputId={inputId} {...props} />;
+});
+ArgumentInput.displayName = 'ArgumentInput';
+
+interface ArrayArgumentInputProps extends Omit<ArgumentInputProps, 'ref'> {
+    inputId: string;
+}
+const ArrayArgumentInput = forwardRef<HTMLInputElement, ArrayArgumentInputProps>(
+    ({ arg, error, value, onChange, onBlur, inputId, ...props }, ref) => {
+        const idCounterRef = useRef(0);
+        const stableIdsRef = useRef<string[]>([]);
+        const inputRefsRef = useRef<Map<string, HTMLInputElement>>(new Map());
+
+        const parseValues = (val: string | number | readonly string[] | undefined): string[] => {
+            if (!val || typeof val !== 'string') return [''];
+            const trimmed = val.trim();
+            if (trimmed === '') return [''];
+            return trimmed.split(',').map(v => v.trim());
+        };
```

Comment:
May I ask to move that helper and maybe wrap logic inside useEffect into a separate hook, please?